### ISSUE TITLE
C++ implementation of policy gradients

### ISSUE THREAD

**Issue by adam-r-kowalski** [pytorch/pytorch#18392]
I have written a reference implementation of the policy gradient reinforcement learning algorithm in Python (PyTorch/TensorFlow), Julia (Flux), and now C++ (PyTorch). I was wondering if you all would like to add it to the examples since it seems like the C++ API doesn't have too many yet for reinforcement learning. If there is anything I can do to clean up / improve my code before hand, I would be happy to do so. If you want to take a look, it is currently living here: https://github.com/adam-r-kowalski/reinforcement_learning/blob/master/policy_gradients/cpptorch/src/main.cc

**Comment by apaszke** [comment]
Thanks for opening the issue! I think it's better to keep the discussion in [the examples repo](https://github.com/pytorch/examples). Also, we'll probably want to wait until we reach stability of the C++ API, because right now it's in beta and is subject to some changes.

**Comment by fmassa** [comment]
Closing following @apaszke comment.
